src/reporters/composite_reporter.js

```diff
@@ -33,24 +33,43 @@ export default class CompositeReporter {
     }
 
     report(span: Span): void {
-        for (let i = 0; i < this._reporters.length; i++) {
-            let reporter = this._reporters[i];
-            reporter.report(span);
-        }
+        this._reporters.map((r) => {
+            r.report(span);
+        });
+    }
+
+    _executeCallbackLast(callback: ?Function): Function {
+        let count = 0;
+        return () => {
+            count++;
+            if (count >= this._reporters.length) {
+                if (callback) {
+                    callback();
+                }
+            }
+        };
+    }
+
+    flush(callback: ?Function): void {
```

Comment:
why does any reporter needs flush()? I see that the only place it's called from is in `Tracer.flush()`, which is not an open-tracing API method. I think it's unnecessary & redundant functionality. We only need close().